packages/navigator/src/browser/navigator-widget.ts

```diff
@@ -79,21 +79,56 @@ export class FileNavigatorWidget extends FileTreeWidget {
                         this.model.expandNode(child);
                     }
                 }
+            }),
+            this.workspaceService.onWorkspaceRootChanged(async event => {
+                if (event) {
+                    await this.refreshWidget(event.root, event.roots, event.workspaceConfigFile);
+                }
             })
         ]);
     }
 
-    protected initialize(): void {
-        this.workspaceService.root.then(async resolvedRoot => {
-            if (resolvedRoot) {
-                const uri = new URI(resolvedRoot.uri);
+    protected async initialize(): Promise<void> {
+        const workspaceChangeData = await this.workspaceService.workspaceRootChange;
+        await this.refreshWidget(workspaceChangeData.root, workspaceChangeData.roots, workspaceChangeData.workspaceConfigFile);
+    }
+
+    protected async refreshWidget(root: FileStat | undefined, roots: FileStat[], workspaceConfig: FileStat | undefined) {
```

Comment:
Would it be better to just rely on `roots` and not `root` at the same time ?
I feel like the prototype should be:
```ts
refreshWidget(roots: FileStat[], workspaceConfig: FileStat | undefined)
```